source/game_sa/LoadedCarGroup.cpp

```diff
@@ -9,42 +9,153 @@
+                    const auto thisModelFrq = CModelInfo::GetVehicleModelInfo(modelId)->m_nFrq;
+                    if (thisModelFrq >= pickedWeight) {
+                        return (eModelID)(modelId);
+                    }
+                    pickedWeight -= thisModelFrq;
+                }
+                NOTSA_UNREACHABLE();
+            }();
+
+            // Check if it's suitable
+            if (   !CTheScripts::HasCarModelBeenSuppressed(pickedModel)
+                && !CTheScripts::HasVehicleModelBeenBlockedByScript(pickedModel)
+                && !CStreaming::WeAreTryingToPhaseVehicleOut(pickedModel)
+                && (!bNotTooManyInTheWorld || CModelInfo::GetVehicleModelInfo(pickedModel)->m_nRefCount <= 2)
+            ) {
+                return pickedModel;
+            }
+        }
+
+        // 10 tries, but no luck
+        return MODEL_INVALID;
+    };
+
+    if (bOnlyPickNormalCars) {
+        // Originally an array was created here, and that was used
+        // That is more performant, but I doubt that this is so performance critical to care about that.
+        return PickRandom(
+            GetAllModels() | rng::views::filter([](int16 modelId) {
+                switch (const auto cls = CModelInfo::GetVehicleModelInfo(modelId)->m_nVehicleClass) {
+                case VEHICLE_CLASS_NORMAL:
+                case VEHICLE_CLASS_POORFAMILY:
+                case VEHICLE_CLASS_RICHFAMILY:
+                case VEHICLE_CLASS_MOTORBIKE:
+                    return true;
+                }
+                return false;
+            })
+        );
+    } else {
+#ifdef FIX_BUGS
+        return PickRandom(GetAllModels());
+#else
+        return PickRandom(m_members);
+#endif
+    }
 }
 
 // 0x611E90
-int32 CLoadedCarGroup::PickLeastUsedModel(int32 minRefs) {
-    return plugin::CallMethodAndReturn<int32, 0x611E90, CLoadedCarGroup *, int32>(this, minRefs);
+eModelID CLoadedCarGroup::PickLeastUsedModel(int32 maxTimesUsed) {
+    if (Empty()) {
+        return MODEL_INVALID;
+    }
+
+    const auto GetMI = [](auto model) { return CModelInfo::GetVehicleModelInfo(model); };
+    const auto ret = rng::min(GetAllModels(), [&](int16 modelA, int16 modelB) {
+        const auto miA = GetMI(modelA), miB = GetMI(modelA);
+        if (miA->m_nRefCount < miB->m_nRefCount) { // Primary sort criteria is `m_nRefCount`
+            return true;
+        }
+        if (miA->m_nRefCount == miB->m_nRefCount) { // If that fails, secondary is `m_nTimesUsed`
+            return miA->m_nTimesUsed < miB->m_nTimesUsed;
+        }
+        return false;
+    });
+
+    if (GetMI(ret)->m_nTimesUsed <= maxTimesUsed) {
+        return (eModelID)(ret);
+    }
+
+    return MODEL_INVALID;
 }
 
 // 0x611C20
-int32 CLoadedCarGroup::GetMember(int32 count) const {
-    return plugin::CallMethodAndReturn<int32, 0x611C20, const CLoadedCarGroup *, int32>(this, count);
+eModelID CLoadedCarGroup::GetMember(int32 idx) const {
```

Comment:
```suggestion
eModelID CLoadedCarGroup::GetMember(uint32 idx) const {
```
idx value cannot be negative